tests/lava/lib/optimization/utils/generators/test_mis.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import unittest
+import networkx as netx
+import numpy as np
+
+from lava.lib.optimization.problems.problems import QUBO
+from lava.lib.optimization.utils.generators.mis import MISProblem, \
+    find_maximum_independent_set, indices_to_binary_vector
+
+
+class TestMISProblem(unittest.TestCase):
+    def setUp(self):
+        self.num_vertices = 15
+        self.connection_prob = 0.75
+        self.seed = 42
+        self.problem = MISProblem(num_vertices=self.num_vertices,
+                                  connection_prob=self.connection_prob,
+                                  seed=self.seed)
+
+    def test_create_obj(self):
+        self.assertIsInstance(self.problem, MISProblem)
+
+    def test_num_vertices_prop(self):
+        self.assertEqual(self.problem.num_vertices, self.num_vertices)
+
+    def test_connection_prob_prop(self):
+        self.assertEqual(self.problem.connection_prob, self.connection_prob)
+
+    def test_seed_prob(self):
+        self.assertEqual(self.problem.seed, self.seed)
+
+    def test_get_graph(self):
+        graph = self.problem.get_graph()
+        self.assertIsInstance(graph, netx.Graph)
+
+    def test_get_graph_matrix(self):
+        matrix = self.problem.get_graph_matrix()
+        self.assertIsInstance(matrix, np.ndarray)
+        self.assertEqual(matrix.shape[0], self.num_vertices)
+        self.assertEqual(matrix.shape[1], self.num_vertices)
+
+    def test_get_complement_graph(self):
+        graph = self.problem.get_complement_graph()
+        self.assertIsInstance(graph, netx.Graph)
+
+    def test_get_complement_graph_matrix(self):
+        matrix = self.problem.get_complement_graph_matrix()
+        self.assertIsInstance(matrix, np.ndarray)
+        self.assertEqual(matrix.shape[0], self.num_vertices)
+        self.assertEqual(matrix.shape[1], self.num_vertices)
+
+    def test_get_as_qubo(self):
+        qubo = self.problem.get_as_qubo(1.0, 4.0)

```

Comment:
I would specifically define w_off and w_diag for clarity.

Reply:
Makes sense.